component/loki/source/aws_firehose/component.go

```diff
@@ -159,7 +161,7 @@ func (c *Component) Update(args component.Arguments) error {
 
 	if err = c.server.MountAndRun(func(router *mux.Router) {
 		// re-create handler when server is re-computed
-		handler := internal.NewHandler(c, c.logger, c.handlerMetrics, c.rbs, newArgs.UseIncomingTimestamp)
+		handler := internal.NewHandler(c, c.logger, c.handlerMetrics, c.rbs, newArgs.UseIncomingTimestamp, string(newArgs.AccessKey))
```

Comment:
I think we need to also set handleNeedsUpdate if the access key changes above.

Reply:
Good catch! I added the check here: https://github.com/grafana/agent/pull/6114/files#diff-85fbd6d7bc146be7386366fda664cc67e006325f5e38a18f4fcc8af4ebb35c63R136-R138
 I also rebased the PR on the latest main version